src/Schema.php

```diff
@@ -920,6 +834,49 @@ protected function getColumnPhpType(ColumnSchemaInterface $column): string
         return parent::getColumnPhpType($column);
     }
 
+    /**
+     * Converts column's default value according to {@see ColumnSchema::phpType} after retrieval from the database.
+     *
+     * @param string|null $defaultValue The default value retrieved from the database.
+     * @param ColumnSchemaInterface $column The column schema object.
+     *
+     * @return mixed The normalized default value.
+     *
+     * @psalm-suppress PossiblyNullArgument
+     */
+    private function normalizeDefaultValue(?string $defaultValue, ColumnSchemaInterface $column): mixed
+    {
+        return match (true) {
+            $defaultValue === null,
+            $column->isPrimaryKey()
+                => null,
+            /** @psalm-var string $defaultValue */
+            in_array($column->getType(), [
+                self::TYPE_TIMESTAMP,
+                self::TYPE_DATE,
+                self::TYPE_TIME,
+            ], true)
+                && in_array(strtoupper($defaultValue), [
+                    'NOW()',
+                    'CURRENT_TIMESTAMP',
+                    'CURRENT_DATE',
+                    'CURRENT_TIME',
+                ], true)
+                    => new Expression($defaultValue),
+            preg_match("/^B?'(.*?)'::/", $defaultValue, $matches) === 1
```

Comment:
You skip this:
elseif ($loadColumnSchema->getType() === 'boolean') {
                    $loadColumnSchema->defaultValue($defaultValue  === 'true');

Reply:
And need add test for check this behaviour 'true' / 'false' as default values. Current tests not cover this.

Behavior in phpTypeCase for default value 't' will be 'true' ater your changes
                return match ($value) {
                    't', 'true' => true,
                    'f', 'false' => false,
                    default => (bool)$value,

Reply:
cases 'true' and 'false' for default values only,
cases 't' and 'f' for array values only,
pdo returns (bool) `true`, `false` or `null`

Added tests

Reply:
First of all thank you for such good PR, in this particular one, I see everything fine, just split the match true in individual ifs, the first gives better readability since there are many chained conditions, the second we would see in the code coverage, which the tests cover each of the conditions,

thks.